test/e2e/istio-csr_test.go

```diff
@@ -0,0 +1,300 @@
+			defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "istio", "proto-cm.yaml"), ns.Name)
+
+			By("applying istio issuer")
+			loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "istio", "istio-issuer.yaml"), ns.Name)
+			defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "istio", "istio-issuer.yaml"), ns.Name)
+
+			By("applying istio-csr resource")
+			loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "istio", "istio-csr.yaml"), ns.Name)
+			defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "istio", "istio-csr.yaml"), ns.Name)
+
+			By("poll till the service account is available")
+			err := pollTillServiceAccountAvailable(ctx, clientset, ns.Name, serviceAccountName)
+			Expect(err).Should(BeNil())
+
+			By("creating jwt token for header authorization")
+			tokenRequest := &authenticationv1.TokenRequest{
+				Spec: authenticationv1.TokenRequestSpec{
+					Audiences: []string{
+						"istio-ca",
+					},
+				},
+			}
+			tokenResponse, err := clientset.CoreV1().ServiceAccounts("istio-system").CreateToken(ctx, serviceAccountName, tokenRequest, metav1.CreateOptions{})
+			Expect(err).Should(BeNil())
+
+			Expect(tokenResponse.Status).ShouldNot(BeNil())
+
+			By("generate csr request")
+			csr, err := generateCSR()
+			Expect(err).Should(BeNil())
+
+			By("creating an grpcurl job")
+			job := &batchv1.Job{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "grpcurl-job",
+				},
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: grpcAppName,
+							Labels: map[string]string{
+								"app": grpcAppName,
+							},
+						},
+						Spec: corev1.PodSpec{
+							RestartPolicy: corev1.RestartPolicyNever,
+							Containers: []corev1.Container{
+								{
+									Name:  grpcAppName,
+									Image: "fullstorydev/grpcurl:latest", // Use your custom image if applicable
+									Args: []string{
+										"-import-path", "/proto",
+										"-proto", "/proto/ca.proto",
+										"-H", fmt.Sprintf("Authorization: Bearer %s", tokenResponse.Status.Token),
+										"-d", fmt.Sprintf(`{"csr": "%s", "validity_duration": 3600}`, csr),
+										"-cacert", "/etc/certs/root-cert.pem",
+										fmt.Sprintf("cert-manager-istio-csr.%s.svc:443", ns.Name),
+										"istio.v1.auth.IstioCertificateService/CreateCertificate",
+									},
+									VolumeMounts: []corev1.VolumeMount{
+										{Name: "cert-volume", MountPath: "/etc/certs"},
+										{Name: "proto", MountPath: "/proto"},
+									},
+								},
+							},
+							Volumes: []corev1.Volume{
+								{
+									Name: "cert-volume",
+									VolumeSource: corev1.VolumeSource{
+										ConfigMap: &corev1.ConfigMapVolumeSource{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: "istio-ca-root-cert",
```

Comment:
does our istio-csr controller add this ConfigMap?
I couldn't spot any resource for `istio-ca-root-cert` in the `istio-system` namespace. Do you need the service mesh installed + istiod running for this (by any chance)?

Doesn't seem the same as whatever you added in the issuer or certificate yaml(s), so am worried :)

Reply:
Nvm, my bad. The istio-csr agent itself copies that config map to each namespace on the cluster. But I think we could poll for that config map to be created first, the Job's Pod stay in ContainerCreating state for a long time until that config map is made available for mounting as a volume.